QUARKUS-2727.md

```diff
@@ -0,0 +1,98 @@
+
+SmallRye Config [has been enhanced](https://github.com/smallrye/smallrye-config/pull/833) with ability to decrypt or decode secret key values.
+Secret key value is resolved from Secret Keys Expression provided by any config source. The expression consist of a Secret Keys Handler name and actual Secret Key.
+Built in interceptor recognize this expression, find the handler by the name and handler transforms Secret Key to actual value.
+There are also handlers provided by the SmallRye Config out of the box (Jasypt, Crypto) that supports basic encryption algorithms.
+
+It is important to verify ability to load secret values from KeyStore as Red Hat Build of Keycloak, Red Hat Process Automation Manager and other products may need this ability.
+In cloud environment, our customers are most likely to store secrets in OpenShift Secrets, and we (Red Hat Build of Quarkus QE team) are best posed to test 
+full application flow, starting in OpenShift Secret and down to the configuration property injected to resource.
+
+## Scope of testing
+
+Tests should verify:
+
+- creating of a custom `SecretKeysHandler` and provide different ways to decode or decrypt configuration
+  values
+- `SecretKeysHandler` registration via the `ServiceLoader` mechanism (both directly and through factory)
+- `SecretKeysHandler` registration to manually built config as we do for Secret Keys Names
+- any config source can contain build time and runtime secret keys expressions:
+  - `application.properties`
+  - password secured KeyStore loaded by KeyStore Config Source (`smallrye-config-source-keystore` dependency) using:
+    - custom handler
+    - out of the box handler
+  - OpenShift secret accessed with Quarkus Kubernetes Config extension as:
+    - file system config secret (`quarkus.openshift.app-secret`)
+    - API server config secret (`quarkus.kubernetes-config.secrets`)
+  - custom config source
+- config profile dependent secret keys expressions
+- using multiple secret keys handlers:
+  - custom secret keys handlers
+  - Jasypt secret keys handler
+  - out-of-the-box handlers provided by Crypto (`smallrye-config-crypto` dependency)
+- modification of secret keys expressions and secret keys handlers in DEV mode
+- secret keys handlers in native mode and make sure all the handlers (custom and built in) are registered for reflection automatically
+- secrets hidden from accidental exposure by `SecretKeysConfigSourceInterceptor`, that is secrets that are also Secret Keys
+- encryption key used to decode secrets encoded by the `AES/GCM/NoPadding` algorithm. The point here is to verify algorithm emphasized in upstream documentation.
+- basic encryption capabilities provided by Jasypt handler
+
+### Existing tests
+
+The Quarkus Test Suite contains Secret Keys Names in its config module.
+Customers are likely to use them together with secrets and so should we.
+Upstream coverage is mostly secured by SmallRye Config project and test coverage inside Quarkus is sporadic.
```

Comment:
> test coverage inside Quarkus is sporadic

Maybe worth creating GH issue for Roberto to enhance the coverage

Reply:
ok, we can keep it that way, just that SR Config is one of the key elements in Quarkus, so that's why I got triggered by that mention :)